### ISSUE TITLE
Support for Managing Project Quotas

### ISSUE THREAD

**Issue by arusso** [terraform-providers/terraform-provider-openstack#657]
Hello,

I would like to be able to manage OpenStack block storage and compute quotas from within Terraform. Current docs suggest this isn't currently possible (or at least I found no mention of them).

gophercloud appears to already have support for this, so I believe this is just a matter of exposing this in the provider:

- [block storage quotasets](https://godoc.org/github.com/gophercloud/gophercloud/openstack/blockstorage/extensions/quotasets)
- [compute quotasets](https://godoc.org/github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/quotasets)

### Terraform Version

```
$ terraform -v
Terraform v0.11.10
+ provider.null v2.0.0
+ provider.openstack v1.14.0
+ provider.vault v1.5.0
```

### Terraform Configuration Files

It seems like this should reasonably be split into two new resources and a  config could look something like:

````
data "openstack_identity_project_v3" "project_1" {
  name = "demo"
}

resource "openstack_compute_quotaset_v2" "project_1-cqs" {
  project = "${data.openstack_identity_project_v3.id}"
  fixed_ips = 50
  keypairs = 100
  cores = 50
  instances = 10
  ...
}

resource "openstack_blockstorage_quotaset_v2" "project_1-bqs" {
  project = "${data.openstack_identity_project_v3.id}"
  snapshots = 20
  gigabytes = 20000
}
```

### Expected Behavior

Quotas should be applied to projects as configured.

### Actual Behavior

NA

**Comment by jtopjian** [comment]
@arusso Thanks!

I think the only difficulty with implementing this is to decide what to do for the "deletion" part. One option is to consider deletion to be a "no-op" and not try to revert any of the previously modified quota items. Another option is to store the original quotas, but that might get a little verbose/ugly.

A note about the compute quotas: though it supports resources like floating IPs and fixed IPs, I don't think those are respected in newer versions of OpenStack. We can still include them to support older versions, but the documentation will need to detail that.

**Comment by arusso** [comment]
I think a no-op on delete is a completely reasonable approach in this
situation.

Including the deprecated quotas is a probably good idea. I know we probably
won’t be on a version that supports the network resource quotas via the
neutron api for a few years so I’d very much like to be able to manage them
until then.

Aaron

**Comment by tibers** [comment]
Please can we have this feature? I was just paging through our codebase and folks are using `local-exec` to set quotas and it's making a mess.

**Comment by ozerovandrei** [comment]
Terraform-provider-openstack v1.21.0 has just been released with the ability to manage Blockstorage (Cinder) V2/V3 project quotas.
https://www.terraform.io/docs/providers/openstack/r/blockstorage_quotaset_v2.html
https://www.terraform.io/docs/providers/openstack/r/blockstorage_quotaset_v3.html

**Comment by thundertaker** [comment]
Are there any news for this issue? We would like to manage Compute quotes by Terraform as well :)

**Comment by ozerovandrei** [comment]
Networking and Compute quotas were also added and they're available in `v1.24.0`.
